### DISCUSSION TITLE
Angular 19 & PrimeNG 17.19 ?

### DISCUSSION THREAD

**Discussion by djay1977** [primefaces/community#3065] [PrimeNG] [answered]
Hello,
Would it be possible to release a version 17.19 of PrimeNG compatible with Angular 19 ?
The migration time to the new version with Design Tokens and TailwindCSS may take us a lot of time.

Thanks in advance

**Comment by AStoker** [comment, marked as answer]
Linking related items:
https://github.com/primefaces/primeng/pull/16783

Angular 19 changed the way the standalone feature works, making libraries compiled in previous versions of angular incompatible with angular 19 projects.

**Reply by djay1977** [reply]
Hello,

In Angular 19, all components are standalone by default.
You just have to declare standalone to false in the component if it's used in a module.

I tested Angular 19 with PrimeNG 17.18.12 and the premium theme Ultima in 17 and everything is functional.

My request is to make version 17 of PrimeNG compatible with 19 and make a 17.19 as was the case for 18.

Why not make a version of PrimeNG 19 compatible with Angular 19 without the new theming?

I notice that there is still a lot of work to finalize the version of PrimeNG with Design Tokens, TailwindCSS, the new Theme designer and update the Premium templates (I use Ultima).
I know that PrimeVue is the framework to realize new developments but I think that they don't rush and do step by step..

I think that it would be reasonable to make a version 19 of PrimeNG compatible with Angular 19 and to take the time to finalize the version with the new theming + update prenium templates in a version of Angular 20.

**Reply by utillity** [reply]
So @AStoker will there be a release so we can switch to ng19 anytime soon?

**Reply by AStoker** [reply]
My mistake for misunderstanding your original question @djay1977. Unfortunately, even with version 18 of PrimeNG, it does not support Angular 19.
Because of the way the builds work with Angular, there would have to be a fair amount of refactoring to get an Angular 19 supported version of PrimeNG 17 (at least from what I saw). As this is the open source world, I don't see a reason somebody couldn't take that task on themselves and try to make a PR.
@utillity, I'm not on the PrimeNG team, so I really don't know. This was just all from my digging into it over a day (because I really wanted to use angular 19, and wanted to know what was broke 😂 ). This is the defect they're using to track adding Angular 19 support: https://github.com/primefaces/primeng/issues/16827
Best bet is to follow it there on any timeline.

**Reply by raboud** [reply]
I would be willing to checkout how to migrate v18 of primeNG to Angular 19 if I had a clue which branch to use...Master, Prod, V18, v18-prod

I've tried v18 with Angular v19 and i get an exception in p-floatlabel.

**Reply by djay1977** [reply]
You just need to publish a 17.19 and specify in the dependencies for Angular: ^17.0.0 || ^18.0.0 || ^19.0.0
The version 17.18.12 works correctly with Angular 19 but not v18 without forgetting all the refactor that it imposes with the Design Tokens, TailwindCSS and the breaking changes

**Reply by awardhan11** [reply]
@djay1977 

Did you do anything different to make PrimeNG version 17.18.12 works with Angular 19 ?
Or just out of the box ?

Thanks,

**Comment by djay1977** [comment]
It works perfectly, just do a "npm i --force" for peer dependencies issues with the Angular version.

**Reply by awardhan11** [reply]
Thanks,...

**Comment by jacobg213** [comment]
Anyone up for reviewing [this PR](https://github.com/primefaces/primeng/pull/18058)? 😀

**Reply by jacobg213** [reply]
Nvm, the PR got rejected.
I published my own version with Angular 19 support on NPM - https://www.npmjs.com/package/@jacobg213/primeng-17. Feel free to use that one.

I don't gurantee any future fixes/updates. I'll only work on it for as long as my project needs to use PrimeNG but anyone is welcome to just grab that fork, fix it up as they need and also release their own version.

I hope this helps someone. 😄
